googletest/src/gtest-port.cc

```diff
@@ -1235,5 +1235,24 @@ const char* StringFromGTestEnv(const char* flag, const char* default_value) {
   return value == NULL ? default_value : value;
 }
 
+// Reads and returns GTEST_OUTPUT and/or XML_OUTPUT_FILE environment
```

Comment:
Ok, restructured this to use StringFromGTestEnv as you mentioned above, and added comments about the relative priority.